pkg/genericreconciler/genericreconciler.go

```diff
@@ -104,6 +106,33 @@ func specDiffers(o1, o2 metav1.Object) bool {
 	return true
 }
 
+func isReferringSameObject(a, b metav1.OwnerReference) bool {
+	aGV, err := schema.ParseGroupVersion(a.APIVersion)
+	if err != nil {
+		return false
+	}
+	bGV, err := schema.ParseGroupVersion(b.APIVersion)
+	if err != nil {
+		return false
+	}
+	return aGV == bGV && a.Kind == b.Kind && a.Name == b.Name
+}
+
+func injectOwnerRefs(o metav1.Object, ref *metav1.OwnerReference) bool {
+	if ref == nil {
+		return false
+	}
+	objRefs := o.GetOwnerReferences()
+	for _, r := range objRefs {
+		if isReferringSameObject(*ref, r) {
```

Comment:
Following the above comment about checking `uid` or not. If 2 ownerRefs have the same group/kind/name but different uid, that means at least one of them doesn't exist anymore. If we compare `uid` in this function, we'd set both as owners -- probably not what we want, because in the case that the original owner is already gone, we want its dependent to be garbage collected with it, even if a new resource (potential owner) with the same name is created later.

Reply:
ack.